flake8_trio/visitors/helpers.py

```diff
@@ -354,16 +352,14 @@ def func_has_decorator(func: cst.FunctionDef, *names: str) -> bool:
 
 
 def get_comments(node: cst.CSTNode | Iterable[cst.CSTNode]) -> Iterator[cst.EmptyLine]:
-    # pyright can't use hasattr to narrow the type, so need a bunch of casts
-    if hasattr(node, "__iter__"):
-        for n in cast("Iterable[cst.CSTNode]", node):
+    if isinstance(node, (cst.CSTNode, cst.MaybeSentinel)):
+        yield from (
+            cst.EmptyLine(comment=ensure_type(c, cst.Comment))
+            for c in m.findall(node, m.Comment())
+        )
+    else:
+        for n in node:
             yield from get_comments(n)
-        return
-    yield from (
-        cst.EmptyLine(comment=ensure_type(c, cst.Comment))
-        for c in m.findall(cast("cst.CSTNode", node), m.Comment())
-    )
-    return
```

Comment:
pyright updated and started requiring even *more* casts (or if that was elsewhere, idr), so rewritten to not require any casts. Although the MaybeSentinel not being in the function header is ... not great